src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -681,24 +718,37 @@ private static object SplitWithPredicate(ExecutionContext context, IScriptExtent
                         // If this delimiter is at the end of the string,
                         // add an empty string to denote the item "after"
                         // it.
-                        if (strIndex == (item.Length - 1))
+                        if (cursor == (item.Length - 1))
                         {
                             split.Add("");
                         }
                     }
                     else
                     {
-                        buf.Append(item[strIndex]);
+                        if(rightToLeft)
+                        {
+                            buf.Insert(0, item[strIndex]);
```

Comment:
We add a char by char here so this is exactly the case when we should avoid `Insert`.
Please here remove `if(rightToLeft) ` and do only `buf.Append(item[strIndex]);` and add in the line 690
```
if(rightToLeft)
{
    buf.Reverse();
}
split.Add(buf.ToString());
buf.Clear();

Reply:
I wanted to do that when we first dicussed the same issue with the `List<string>`, but unfortunately there's no such thing as `StringBuilder.Reverse()` :-\

Reply:
I think we can replace  `StringBuilder` with `List<Char>`.